### ISSUE TITLE
[bug] rule severity string "off" is truthy — does not disable a rule (only boolean false does)

### ISSUE THREAD

**Issue by vlad-ryzhkov** [zernie/agent-lint#112]
Setting `"require-instructions-spec": "off"` in `.vigilesrc.json` `rules` does not silence the warning; `lint` still prints it for `CLAUDE.md` / `AGENTS.md`.

**Root cause (traced).** The `rules` map _is_ read and _does_ gate the rule — so this isn't "the map is ignored". The real footgun: the rule checks `if (specSeverity && …)` (`src/core/validate.ts:296-310`), and the JS string `"off"` is truthy, so the rule still fires. Only the JSON boolean `false` short-circuits it. `docs/rules/require-instructions-spec.md` documents only `"error"` / `"warn"` / `false` — `"off"` is not an accepted value, and the `"off"`-string path is untested (`validate.test.ts` always disables via `false`). Note `rule-meta.ts` already treats `"off"` as the _display_ synonym of `false`, so accepting it as input is the natural expectation an ESLint user brings.

**Expected:** normalize the `"off"` string (and `0`) to `false` when loading config, so it disables the rule like ESLint's `"off"`.

**Suggested fix.** In `loadConfig()` (`src/core/validate.ts:152-188`), normalize severity values before the `rules: { ...DEFAULT_RULES, ...userConfig.rules }` merge (L168):

```ts
function normalizeSeverity(v: unknown): unknown {
  if (v === "off" || v === 0) return false;
  if (Array.isArray(v)) return [normalizeSeverity(v[0]), v[1]];
  return v;
}
```

Applied per-key to `userConfig.rules`, this fixes `"off"` for every rule at once (they all funnel through the merged `config.rules`).

**Comment by zernie** [comment]
Fixed — and thanks for the traced root cause (truthy `"off"` string, not "the map is ignored"). `loadConfig` now runs a `normalizeSeverity` over `userConfig.rules` before the merge (`src/core/validate.ts`), mapping `"off"` and `0` (and array forms) to `false`, so `"off"` disables a rule like ESLint's — for every rule at once, exactly as you suggested. The `"off"`-path is covered by a test now too. (Shipped a while back; the issue just wasn't auto-closed — closing now.)

---
_Generated by [Claude Code](https://claude.ai/code)_
